### PR TITLE
Enable mode 2 ASLR on Linux.

### PR DESCRIPTION
**Pull request by vext01**
Since the virtual addresses of a VM's data can be considered an internal
source of variation.

Also removed recursion, simplified, and tested.

OK?

### CODE REVIEW

File: krun/platform.py [hunk 1]
```diff
@@ -1057,26 +1058,26 @@ def _check_rmsr_loaded(self):
...
         debug("rmsr loaded successfully.")
 
-    def _check_aslr_disabled(self):
-        debug("Checking ASLR is off")
-        with open(self.ASLR_FILE, "r") as fh:
-                enabled = fh.read().strip()
-        if enabled == "0":
-            return  # OK!
-        else:
-            # ASLR is off, but we can try to enable it
```
Comment (vext01) [inline krun/platform.py]: This comment was outright wrong by the way.

Comment (snim2) [inline krun/platform.py]: Yup!

File: krun/platform.py [hunk 2]
```diff
@@ -1057,26 +1058,26 @@ def _check_rmsr_loaded(self):
...
-            debug("Turning ASLR off")
-            cmd = "%s sh -c 'echo 0 > %s'" % \
-                (self.change_user_cmd, self.ASLR_FILE)
-            stdout, stderr, rc = run_shell_cmd(cmd, failure_fatal=False)
+    def _check_aslr_enabled(self):
+        """Check ASLR is cranked to level 2"""
```
Comment (snim2) [inline krun/platform.py]: Maybe say what `2` means:

> 0 – No randomization. Everything is static.
>1 – Conservative randomization. Shared libraries, stack, mmap(), VDSO and heap are randomized.
> 2 – Full randomization. In addition to elements listed in the previous point, memory managed through brk() is also randomized.

Comment (vext01) [inline krun/platform.py]: Where is that text from, or is it original?

File: krun/platform.py [hunk 3]
```diff
@@ -1057,26 +1058,26 @@ def _check_rmsr_loaded(self):
...
-                fatal(msg)
+        debug("Checking ASLR is on")
+        for rep in 1, 2:
+            with open(LinuxPlatform.ASLR_FILE, "r") as fh:
+                val = fh.read().strip()
+            if val == str(LinuxPlatform.ASLR_MODE):
+                return  # OK!
             else:
-                self._check_aslr_disabled()  # should work this time
+                if rep == 2:
```
Comment (snim2) [inline krun/platform.py]: I guess there isn't an easy way to test these failure modes?

Comment (vext01) [inline krun/platform.py]: Not easily, I could split out the opening of the file into a new method which could be monkeypatched. Is it worth the effort?

File: krun/platform.py (hunk 2 above)
Comment (snim2) [inline krun/platform.py]: StackOverflow. You probably want to quote this file: https://www.kernel.org/doc/Documentation/sysctl/kernel.txt (grep for `randomize_va_space`)

File: krun/platform.py (hunk 3 above)
Comment (snim2) [inline krun/platform.py]: No, probably not

File: krun/platform.py (hunk 2 above)
Comment (vext01) [inline krun/platform.py]: Fixed.

**Comment by snim2** [comment]
OK, nice! Do you want to squash?

**Comment by vext01** [comment]
splat

**Pull request merged**